src/components/modules/api/blocks.ts

```diff
@@ -10,13 +10,31 @@ import BlockAPI from '../../block/api';
  * provides with methods working with Block
  */
 export default class BlocksAPI extends Module {
+  /**
+   * Method names that must be decorated
+   */
+  protected decorateList: string[] = [
+    'clear',
+    'render',
+    'renderFromHTML',
+    'delete',
+    'swap',
+    'move',
+    'getBlockByIndex',
```

Comment:
this method can work in read-only

Reply:
Then we need to disable Block API methods in read-only as well